spec/tbd_tests_spec.rb

```diff
@@ -1566,3 +1566,414 @@
+          "items": {
+            "$ref": "#/definitions/Edge"
+          },
+          "uniqueItems": true
+        },
+        "surfaces": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Surface"
+          },
+          "uniqueItems": true
+        },
+        "spaces": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Space"
+          },
+          "uniqueItems": true
+        },
+        "stories": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Story"
+          },
+          "uniqueItems": true
+        },
+        "unit": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Unit"
+          },
+          "uniqueItems": true
+        },
+        "logs": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Log"
+          }
+        }
+      },
+      "additionalProperties": false,
+      "definitions": {
+        "PSI": {
+          "description": "Set of PSI-values (in W/K per m) for thermal bridges",
+          "type": "object",
+          "properties": {
+            "id": {
+              "title": "Unique PSI set identifier",
+              "type": "string"
+            },
+            "rimjoist": {
+              "title": "Floor/wall edge PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "parapet": {
+              "title": "Roof/wall or exposed-floor/wall edge PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "fenestration": {
+              "title": "Window, door, skylight perimeter PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "concave": {
+              "title": "Wall corner [0°,135°) PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "convex": {
+              "title": "Wall corner (225°,360°] PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "balcony": {
+              "title": "Floor/balcony edge PSI ",
+              "type": "number",
+              "minimum": 0
+            },
+            "party": {
+              "title": "Party wall edge PSI",
+              "type": "number",
+              "minimum": 0
+            },
+            "grade": {
+              "title": "Floor/foundation edge PSI",
+              "type": "number",
+              "minimum": 0
+            }
+          },
+          "additionalProperties": false,
+          "required": [
+            "id"
+          ],
+          "minProperties": 2
+        },
+        "KHI": {
+          "description": "KHI-value (in W/K) for a point thermal bridge",
+          "type": "object",
+          "properties": {
+            "id": {
+              "title": "Unique KHI identifier",
+              "type": "string"
+            },
+            "point": {
```

Comment:
Would you want to enumerate the types of KHI beyond just point?  The types of PSI correspond to real world things.

Reply:
That's a good question. In short "no", but I think it's best to state why as we may want to revisit this. So true, enumerated PSI types do correspond to real world things. But more importantly, they directly relate to keywords in current standards and codes, which often reserve particular treatment for each thermal bridge (TB) type. So the enumerated PSI types are key to (automatically) generating _valid_ reference/budget buildings. For instance, pre-2017 Canadian energy codes offer exceptions to slab-wall edges ("rimjoist"), while the QC energy code has different requirements for slab-wall ("rimjoist"), roof-wall ("parapet"), and slab projections ("balcony"). The "fenestration" type could one day (?) be expanded (as an option) to differentiate between "jamb", "sill" and "lintel" - but Industry isn't there yet, so there's little value in pre-emptively expanding TB types IMO. For _point_ TBs, their current treatment is quite coarse - I don't see any differentiation in codes/standards between a series of columns punching through an exposed floor vs roof supports for HVAC equipment. For instance, the QC energy code has a default 0.5 W/K requirement for any major _point_ TB, regardless of application. It's better than nothing, but I expect more granularity in the future. Similarly to fenestration vs jamb-sill-lintel, once someone can point me to a code or standard that differentiates between point TB types, I'll gladly add that in. In the meantime, TBD doesn't hold enumerated KHI types.